# **SYNTAX**

- **Syntax error** means variable properly not initialized
    - **example: number one=100**
 
- **Name error** means variable not initialized but we're trying to use it

    - **example: name not defined**

- **Using inbuilt functions** as variables

    - It is not recomended to use inbuilt functions as variables
    - Inbuilt functions as special use case

- **Using keywords** as variables

    - It is not recomended to use keywords as variables
    - Keywords are also have special purpose use in python

In [3]:
sum([10,20,30])

60

In [7]:
str1='hello"

SyntaxError: unterminated string literal (detected at line 1) (1010079629.py, line 1)

# **Unterminated string literal**

    - Focus on whether we are closing string properly or not

    - Do not use triple quotes when you're using a string for code purpose

    - Triple quotes **Doc string**  or **Multi-line comment**

In [13]:
int('10.5')

ValueError: invalid literal for int() with base 10: '10.5'

# **Type-casting Error's**
- **Value error** converting string representation of float into integer

  - **example: int('10.5')**

- Always be cautious when boolean will give **True** and **False**

bool(10)
bool(-10)
bool(0)
bool(10.6)
bool(-10.6)
bool(0.0)
bool('apple')
bool('0')
bool('')

- What is the use of type : type([10,20,30])

- What is the use of len : len([10,20,30])

- What is the use of sum : sum([10,20,30])

- What is the use of max : max([10,20,30])

- What is the use of min : min([10,20,30])

In [25]:
type([10,20,30])

list

In [27]:
len([10,20,30])

3

In [29]:
sum([10,20,30])

60

In [31]:
max([10,20,30])

30

In [33]:
min([10,20,30])

10

type()
len()
max()
min()
print()
eval()
input()
int()
float()

In [36]:
input

<bound method Kernel.raw_input of <ipykernel.ipkernel.IPythonKernel object at 0x000001C487CD7890>>

- inbuilt functions expects paranthesis
    
    -  if brackets are missed it will give indication as

    -  **bound method** or **function**

In [42]:
a=100
print('the value of a is :',a)
print('the value of a is : {}'.format(a))
print(f'the value of a is : {a}')

the value of a is : 100
the value of a is : 100
the value of a is : 100


# **END Operator**

In [46]:
print(10,end=' ')
print(20)

10 20


In [49]:
print(10,end='-->')
print(20)

10-->20


In [57]:
print("Hello",end=' ')
print("Hi")

Hello Hi


# **SEPERATOR**

In [63]:
print(10,20)   # 10-->20
print(10,20,sep='-->')

10 20
10-->20


# **Input**

In [69]:
city='Hyd'
city=input('Enter your city: ')

Enter your city:  Delhi


In [71]:
input()

 10


'10'

In [75]:
int(input()) # int('10')

 10


10

In [77]:
float(input())  # float('10')

 10


10.0

In [79]:
eval(input())

 10.5


10.5

In [81]:
26/5

5.2

In [83]:
26//5

5

In [85]:
26%5

1

- 26/5 #Normal division 5.2
- 26//5 # floor division : Quotient
- 26%5 # Modulus operator : Remainder